jsf-ri/src/main/java/com/sun/faces/spi/AnnotationProviderFactory.java

```diff
@@ -70,11 +70,7 @@ public static AnnotationProvider createAnnotationProvider(ServletContext sc) {
         String[] services = ServiceFactoryUtils.getServiceEntries(ANNOTATION_PROVIDER_SERVICE_KEY);
         if (services.length > 0) {
             // only use the first entry...
-            Object provider = ServiceFactoryUtils.getProviderFromEntry(services[0],
```

Comment:
I don't think this is the right fix. The [JavaDoc](https://github.com/eclipse-ee4j/mojarra/blob/2.3/impl/src/main/java/com/sun/faces/spi/AnnotationProvider.java#L64-L77) for ```AnnotationProvider``` says:

```
Custom AnnotationProviders can be used to decorate the default AnnotationProvider
by providing a constructor that takes an AnnotationProvider as the second parameter:

public AnnotationProvider(ServletContext sc, AnnotationProvider parent)

If decoration is not desired, then the custom provider must have a constructor that takes one parameter, a
ServletContext:

public AnnotationProvider(ServletContext sc)
```

Since providing a constructor with two parameters is allowed, I don't think it makes sense to remove this. One option might be to just catch the ```NoSuchMethodException``` that's thrown by ```getProviderFromEntry``` and try using one parameter instead.

Reply:
> catch the NoSuchMethodException that's thrown by getProviderFromEntry and try using one parameter instead.

This is the current behavior, excepts the exception is caught in https://github.com/jboss/mojarra/blob/2.3.9.SP/impl/src/main/java/com/sun/faces/spi/ServiceFactory.java#L64-L69 and log a trace level message, and return null.

```
        } catch (ClassNotFoundException | NoSuchMethodException | SecurityException | FacesException | InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
            if (LOGGER.isLoggable(Level.FINE)) {
                LOGGER.log(Level.FINE, e.toString(), e);
            }
            return null;
}
```
After that, it continues to try with one parameter instead.

Reply:
@fjuma Back to the issue report, the exception is only logged when people turn on TRACE logging for javax.enterprise.resource.